src/nvim/eval.c

```diff
@@ -5985,7 +5985,7 @@ void timer_due_cb(TimeWatcher *tw, void *data)
   // Handle error message
   if (called_emsg > called_emsg_before && did_emsg) {
     timer->emsg_count++;
-    if (current_exception != NULL) {
+    if (did_throw) {
```

Comment:
This is probably not what the code here is intended to do. The corresponding code in Vim (<https://github.com/vim/vim/commit/c577d813b7978345dec4310b2d8f5d5624a681f6>) saves `did_throw` and doesn't discard exception if `did_throw` is previously set. Not sure if this can happen in Nvim though, as Nvim processes timers on the main loop.

Reply:
Oh, timers can be processed from `getchar()`. I'll try to check what should be done there.

Reply:
I still cannot find a place where `did_throw` can be `true` when `timer_due_cb()` is called.